pkg/operator/metrics/metrics.go

```diff
@@ -0,0 +1,21 @@
+package metrics
+
+import (
+	"time"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+var expiryTime = metrics.NewGauge(&metrics.GaugeOpts{
+	Name: "service_ca_expiry_time_seconds",
+	Help: "Reports the expiry time of service CA used in the cluster.",
+})
+
+func init() {
+	legacyregistry.MustRegister(expiryTime)
```

Comment:
just a nit: instead of having a global registry, maybe think about dependency injecting the prometheus registry.